rust/foxglove-derive/src/lib.rs

```diff
@@ -52,12 +52,7 @@ fn derive_enum_impl(input: &DeriveInput, data: &DataEnum) -> TokenStream {
             }
 
             fn write(&self, buf: &mut impl ::foxglove::bytes::BufMut) {
-                let mut value = *self as u64;
-                while value >= 0x80 {
-                    buf.put_u8((value as u8) | 0x80);
-                    value >>= 7;
-                }
-                buf.put_u8(value as u8);
+                ::foxglove::encode_varint(*self as u64, buf);
```

Comment:
maybe worth namespacing all the custom protobuf encoding stuff into a `protobuf` namespace?

Or another trick I've seen recently - the `subtle` namespace:

https://v8docs.nodesource.com/node-20.3/d1/d40/namespacecppgc_1_1subtle.html

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle

Reply:
Done (as `protobuf`)